integration-cli/docker_api_containers_test.go

```diff
@@ -1948,3 +1949,28 @@ func (s *DockerSuite) TestContainerKillCustomStopSignal(c *check.C) {
 	err = waitInspect(id, "{{.State.Running}} {{.State.Restarting}}", "false false", 30*time.Second)
 	c.Assert(err, checker.IsNil)
 }
+
+func (s *DockerSuite) TestContainerAPIArchiveWithHardlink(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	containerID := makeTestContainer(c, testContainerOptions{addContent: true})
+
+	tmpDir := getTestDir(c, "test-cp-from-dir-with-hardlink")
+	defer os.RemoveAll(tmpDir)
+
+	srcPath := "/dir6"
+	dstPath := cpPath(tmpDir, "dir6_")
+
+	res, body, err := request.SockRequestRaw("GET", fmt.Sprintf("/containers/%s/archive?path=%s", containerID, srcPath), nil, "application/x-tar", daemonHost())
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
+
+	srcInfo := archive.CopyInfo{
+		Path:   srcPath,
+		Exists: true,
+		IsDir:  true,
+	}
+	err = archive.CopyTo(body, srcInfo, dstPath)
+	c.Assert(err, checker.IsNil)
+	c.Assert(fileContentEquals(c, cpPath(tmpDir, "dir6_/hardLinkFile1"), "dir6/hardLinkFile1\n"), checker.IsNil)
```

Comment:
this code test the following case,
```
docker cp id:/dir6  /tmpxxx/_dir6
```
"/dir6_/hardLinkFile1" is in container, we can't use it to compare. My goal is testing the issue located in https://github.com/moby/moby/pull/33956,and if the file have the exact content, I think we don't need compare  os.FileInfo ,  -_-  .